oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/AuthorizationResponse.java

```diff
@@ -50,19 +50,27 @@ public String getState() {
     }
 
     public static AuthorizationResponse fromString(final String s) throws AuthorizationException {
+        return fromString(s, null);
+    }
+
+    public static AuthorizationResponse fromString(final String s, final String potentialDescription) throws AuthorizationException {
         String code = null;
         String state = null;
         String error = "unknown_error";
-        String errorDescription = null;
+        String errorDescription = potentialDescription;
         String errorUriString = null;
-        if (s != null) {
+        if (s != null && s.length() > 0) {
             final String[] pairs = PAIR_SEPARATOR.split(s.trim());
 
             for (final String pair : pairs) {
                 final String[] nameAndValue = NAME_VALUE_SEPARATOR.split(pair, 2);
                 try {
                     if (nameAndValue.length != 2) {
-                        throw new AuthorizationException("parsing_error", "Failed to parse server response", null, null);
+                        final StringBuilder sb = new StringBuilder("Failed to parse server response");
+                        if (errorDescription != null) {
+                            sb.append(".\nDetails: ").append(errorDescription);

```

Comment:
I don't oppose this change, but be noted that this is not true -- the stderr have nothing to do with server output (in the Ubuntu box where this issue was found) and you are assigning the error as a "detail" to the parsing failure -- which is a legit server failure.  

It's confusing to consumer at the least.  I prefer exception chaining in this case since those two are really distinct exceptions.